invenio_records_rest/config_loader.py

```diff
@@ -0,0 +1,298 @@
+
+    @cached_property
+    def delete_permission_factory(self):
+        return self.delete_permission_factory_imp or \
+            current_records_rest.delete_permission_factory
+
+    @cached_property
+    def read_permission_factory(self):
+        return self.read_permission_factory_imp or \
+            current_records_rest.read_permission_factory
+
+    @cached_property
+    def record_loaders(self):
+        return self.endpoint_record_loaders or \
+            current_records_rest.loaders
+
+
+class DictField(fields.Field):
+    """Marshmallow dictionary field used when field names are unknown."""
+    def __init__(self, key_field, nested_field, *args, **kwargs):
+        fields.Field.__init__(self, *args, **kwargs)
+        self.key_field = key_field
+        self.nested_field = nested_field
+
+    def _deserialize(self, value, attr, data):
+        result = {}
+        self._validate_missing(value)
+        for key, subvalue in six.iteritems(value):
+            deserialized_key = self.key_field.deserialize(key)
+            deserialized_value = self.nested_field.deserialize(subvalue)
+            result[deserialized_key] = deserialized_value
+        return result
+
+
+class CompletionConfig(fields.Field):
+    """Record REST endpoint's completion suggester configuration schema."""
+
+    field = fields.Str(required=True)
+
+    context = fields.Str(required=False)
+
+    size = fields.Integer(required=False)
+
+
+class SuggesterConfig(fields.Field):
+    """Record REST endpoint's suggester configuration schema."""
+
+    completion = fields.Nested(CompletionConfig, required=False)
+
+
+def deserialize_import_string_dict(obj):
+    """Deserialize dicts of string->import_string."""
+    if not isinstance(obj, dict):
+        raise ValidationError(
+            'Field "{}" should be an import string.'.format(attr),
+            field_names=[attr]
+        )
+    return {key: obj_or_import_string(obj)
+            for key, obj in six.iteritems(obj)}
+
+
+class EndpointConfigSchema(Schema):
+    """Record REST Endpoint Configuration schema."""
+
+    @post_load
+    def build_config(self, data):
+        """Finish configuration loading."""
+        search_class_kwargs = {}
+        if data.get('search_index'):
+            search_class_kwargs['index'] = data['search_index']
+        else:
+            data['search_index'] = data['search_class'].Meta.index
+
+        if data.get('search_type'):
+            search_class_kwargs['doc_type'] = data['search_type']
+        else:
+            data['search_type'] = data['search_class'].Meta.doc_types
+
+        if search_class_kwargs:
+            data['search_class'] = partial(data['search_class'],
+                                           **search_class_kwargs)
+
+        data['resolver'] = Resolver(
+            pid_type=data['pid_type'],
+            object_type='rec',
+            getter=partial(data['record_class'].get_record, with_deleted=True))
+        return EndpointConfig(**data)
```

Comment:
Do we need to wrap the data to another class? Can't we have all the "cached properties" as special fields on schema?